libs/go/libp2p_node/aea/api.go

```diff
@@ -388,6 +395,29 @@ func (aea *AeaApi) listen_for_envelopes() {
 	}
 }
 
+func (aea *AeaApi) envelopeSendLoop() {
+	logger.Debug().Msg("send loop started")
+	var err error
+	for {
+		envelope := <-aea.send_queue
+		logger.Debug().Msg("send lopp: got envelope")
```

Comment:
```suggestion
		logger.Debug().Msg("send loop: got envelope")
```

Reply:
status consumed by the one and only goroutine - sending loop and populated by receive loop.
so there is can be a deadlock if protocol goes wrong, probably need to flush queue if it's not empty on next status received